scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DatasetService.java

```diff
@@ -156,4 +156,13 @@ public boolean deleteFileFromDataset(Long datasetId, Long fileId) throws Dataset
             throw new DatasetStorageException("File was created, but could not be deleted", e);
         }
     }
+
+    @Override
+    public Integer getSize(Long datasetId) {
+        Optional<Dataset> dataset = datasetRepository.findById(datasetId);
```

Comment:
There should be a count method to get the number of results. You don't want to load them all into memory just to count them. It might be thousands. Let the db handle that.

Also, I would return long. There might be many files in a dataset.